src/pubsub.c

```diff
@@ -615,6 +615,116 @@ void publishCommand(client *c) {
     addReplyLongLong(c,receivers);
 }
 
+typedef struct {
+    dict *subscription_dict;
+    long count;
+    int use_pattern;
+    sds pat;
+} pubsubSubscribersCmdArgs;
+
+static inline void pubsubSubscribersCmdArgsInitializeDefault(pubsubSubscribersCmdArgs *args) {
+    args->count = 1000;
+    args->use_pattern = 0;
+}
+
+static void pubsubReplySubscribersClientInfo(client *c, list *clients) {
+    listNode *ln;
+    listIter li;
+    listRewind(clients, &li);
+    while ((ln = listNext(&li)) != NULL) {
+        client *client = listNodeValue(ln);
+        addReplyMapLen(c, 3);
+        addReplyBulkCString(c, "name");
+        if (client->name) {
+            addReplyBulk(c, client->name);
+        } else {
+            addReplyNull(c);
+        }
+        addReplyBulkCString(c, "id");
+        addReplyBulkLongLong(c, client->id);
+        addReplyBulkCString(c, "addr");
+        addReplyBulkCString(c, getClientPeerId(client));
+    }
+}
+
+/* Parse the arguments for PUBSUB SUBSCRIBERS command */
+static int parseArgsPubSubSubscribersCmd(client *c, pubsubSubscribersCmdArgs *args) {
```

Comment:
```suggestion
static int parseArgsPubSubSubscribersCmdOrReply(client *c, pubsubSubscribersCmdArgs *args) {
```
Is the more canonical form.